### ISSUE TITLE
Verifying buffer for zero copy routines

### ISSUE THREAD

**Issue by tammyleino** [OpenAMP/open-amp#360]
The following API functions take a buffer address as an input parameter but do not validate that the buffer is valid; rpmsg_hold_rx_buffer, rpmsg_release_rx_buffer, rpmsg_send_offchannel_nocopy.

In forming a solution, I have a question about the virtqueue buffers.  Will all the buffers in the virtqueue always be of the same size or are there implementations that allocate buffers on the fly of varying sizes?

**Comment by arnopo** [comment]
Right, a test is done to check that buffer is not null but not that the buffer is valid.  Checking that the buffer is in the shared memory prevents from writing in other memory but not prevent to corrupt other buffers or vrings.

On the other hand , to many checks on buffer would impact the performance.

for this issue, a pull request would allow a better understanding of the impact.

**Comment by tammyleino** [comment]
https://github.com/OpenAMP/open-amp/pull/406
